querier.go

```diff
@@ -54,6 +54,9 @@ type Series interface {
 
 	// Iterator returns a new iterator of the data of the series.
 	Iterator() SeriesIterator
+
+	// Chunks returns a copy of the compressed chunks that make up this series.
+	Chunks() []chunks.Meta
```

Comment:
Similar to the comment above, `Chunks` is breaking the consistency here a bit, becaue we have `Iterator` above so maybe `ChunksIterator` would be more appropriate? and have `ChunkIterator` iterator that iterates over chunks.

I know iterator interface is sometimes bit controversial (e.g very hard to read and debug), but especially for remote read use case it might be actually more valuable to have it in `iterator`. This is to so potentially we don't need to allocate the whole slice or chunks within `Series interface implementation` once we compose the frame. For cortex use case, I am not sure as you just mentioned: `This is to enable Cortex to use TSDB`  What do you think?

> of the compressed chunks 

Does it mean we guarantee ALL implementations to return chunks with `chunks.Meta.Bytes` and not only `Ref`? I don't think we can guarantee that so we might want to add `or ref to chunk in chunk file`

Reply:
Proposed iterators here: https://github.com/prometheus/tsdb/pull/665 (: Let me know if this makes sense.

Reply:
I think the iterator approach is better.